python/ray/llm/tests/serve/cpu/builders/test_application_builders.py

```diff
@@ -201,6 +201,23 @@ def test_build_llm_deployment_with_name_prefix(
         handle = serve.run(app)
         assert handle.deployment_name.startswith(_name_prefix_for_test)
 
+    def test_build_llm_deployment_name_prefix_overriden_by_deployment_config(
```

Comment:
I don't think this test is the right requirement. You want the name to not get overriden but prefix would have its own part in the final name.